orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py

```diff
@@ -0,0 +1,132 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import sys
+from torch.onnx import symbolic_helper
+
+
+def _export(g, n, *args, **kwargs):
+    try:
+        name = kwargs['name']
+        inplace = kwargs['inplace']
+        training_mode = symbolic_helper._training_mode
+        cconv = n.cconv()
+        input_tensor_types = []
+        input_tensor_requires_grads = []
+
+        input_int_scalars = []
+        input_int_scalar_positions = []
+
+        input_float_scalars = []
+        input_float_scalar_positions = []
+
+        input_int_tuples = []
+        input_int_tuple_positions = []
+        input_int_tuple_begins = []
+
+        input_float_tuples = []
+        input_float_tuple_positions = []
+        input_float_tuple_begins = []
+
+        input_pointer_scalars = []
+        input_pointer_scalar_positions = []
+
+        tensor_args = []
+        for i, arg, call_type in zip(range(len(args)), args, cconv):
+            if call_type == 'd':
+                # Got a tensor variable.
+                tensor_args.append(arg)
+
+                requires_grad = 1 if arg.requires_grad() else 0
+                input_tensor_requires_grads.append(requires_grad)
+
+                scalar_type = int(symbolic_helper.cast_pytorch_to_onnx[arg.type(
+                ).scalarType()])
+                input_tensor_types.append(scalar_type)
+            elif call_type == 'c':
+                # Got a non-tensor variable.
+                # Non-tensor can't have gradient.
+                input_tensor_requires_grads.append(0)
+                if isinstance(arg, float):
+                    # A float.
+                    input_float_scalar_positions.append(i)
+                    input_float_scalars.append(arg)
+                elif isinstance(arg, int):
+                    # A int.
+                    input_int_scalar_positions.append(i)
+                    input_int_scalars.append(arg)
+                elif isinstance(arg, tuple):
```

Comment:
We haven't see a case with empty tuple, so I think no. We support empty list easily if we really want. I will add an assert to ensure the tuple is not empty because this case looks like a bug.